docker/docker.go

```diff
@@ -52,6 +45,17 @@ func GetClient() (*client.Client, error) {
 			return nil, err
 		}
 
+		logrus.Debug("Checking for Docker Toolbox")
+		// The following `SplitN` is safe as the `c.DaemonHost()` has been
+		// already checked when creating the client
+		scheme := strings.SplitN(c.DaemonHost(), "://", 2)[0]
+		// If scheme is `tcp` it means that the host is using a VM
+		// In case of Docker Desktop for Unix this is `unix`
+		// In case of Docker Desktop for Windows this is `npipe`
+		if scheme == "tcp" {
```

Comment:
this check is also incorrect. Native docker [can use](https://docs.docker.com/engine/reference/commandline/dockerd/#daemon-socket-option) tcp too, it just isn't default option.

Reply:
Thank you! I've just read about all the options. Ok, by digging more I think that the only reliable information is [this](https://docs.docker.com/engine/api/v1.39/#operation/SystemInfo), in particular, the `OperatingSystem `.

In my case I have:
```
Operating System: Boot2Docker 18.09.4 (TCL 8.2.1)
```
and:
```
Operating System: Docker for Mac
```

So I guess that I could do something like `strings.Contains(os, "boot2docker")`. WDYT?

Reply:
if this is the only option we can differentiate "native" docker from docker-toolbox/docker-machine.

Reply:
I've tested also on Windows and the `Operating System` contains `Boot2Docker` when using Docker Toolbox and `Docker for Windows` when natively, similarly to OSX.

Reply:
Addressed [here](https://github.com/src-d/engine/pull/428/commits/6399fa108da6f1216b2f38f3fe24a1262d81f2a0).